Feature: Block Strength

Thunderblade73 opened the pull request:
## Dependencies
- https://github.com/hannibal002/SkyHanni-REPO/pull/270

## What
Shows how many ticks you need to break any block with your mining speed.

It opens the skyblock menu to collect up to date stats.

<details>
<summary>Images</summary>

![image](https://github.com/user-attachments/assets/140639b2-dc0b-4308-82dd-57bfa45dae46)

</details>

## Changelog New Features
+ Added /shblockstrength. - Thunderblade73
    * Displays how many ticks are needed to break any block with your mining speed.

## Changelog Technical Details
+ Added Renderable.renderAndScale. - Thunderblade73
+ Added the title of the Inventory to InventoryCloseEvent. - Thunderblade73

Thunderblade73 committed:
init

Thunderblade73 committed:
deleted debug

qtlunya commented:
One small issue, this won't be fully accurate with HOTM perks and tools that give extra Mining Speed only for specific types of blocks. This could probably be read from HotmAPI and the held tool's lore.

CalMWolfs committed:
Merge branch 'beta' into block-break

Thunderblade73 committed:
account for special speeds

Thunderblade73 committed:
Merge remote-tracking branch 'origin/block-break' into block-break

Thunderblade73 commented:
> One small issue, this won't be fully accurate with HOTM perks and tools/pets that give extra Mining Speed only for specific types of blocks. This could probably be read from HotmAPI and then hardcode known cases for tools/pets (or try to read it out from their lore).

I added a way to account for them now. I only implemented HOTM perks since I don't know other sources of specific speed.

Thunderblade73 committed:
some detekt fixes

Thunderblade73 committed:
added more sources

Thunderblade73 commented:
> > One small issue, this won't be fully accurate with HOTM perks and tools/pets that give extra Mining Speed only for specific types of blocks. This could probably be read from HotmAPI and then hardcode known cases for tools/pets (or try to read it out from their lore).
> 
> I added a way to account for them now. I only implemented HOTM perks since I don't know other sources of specific speed.

Empa gave me a full list. It does now account for lapidary, Gemstone drills as well. And I also added a todo with the remaining sources.

Thunderblade73 committed:
Merge remote-tracking branch 'refs/remotes/master/beta' into block-break

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/features/mining/OreBlock.kt
#	src/main/java/at/hannibal2/skyhanni/utils/InventoryUtils.kt

Thunderblade73 committed:
detekt

Thunderblade73 committed:
detekt2

Thunderblade73 committed:
Merge remote-tracking branch 'refs/remotes/master/beta' into block-break

Thunderblade73 committed:
rename of json

Thunderblade73 committed:
Merge remote-tracking branch 'origin/block-break' into block-break

Thunderblade73 committed:
Merge remote-tracking branch 'refs/remotes/master/beta' into block-break

Thunderblade73 committed:
fix compile error

CalMWolfs committed:
Merge branch 'beta' into block-break

CalMWolfs committed:
Merge branch 'refs/heads/beta' into fork/Thunderblade73/block-break

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/features/mining/OreBlock.kt

CalMWolfs committed:
fix merge (i hate detekt)

CalMWolfs committed:
Merge branch 'beta' into block-break

CalMWolfs committed:
fix merge

CalMWolfs committed:
Merge branch 'beta' into block-break

Thunderblade73 committed:
Merge remote-tracking branch 'master/beta' into block-break

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/utils/InventoryUtils.kt

Thunderblade73 committed:
Merge remote-tracking branch 'master/beta' into block-break

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/data/model/SkyblockStat.kt

Thunderblade73 committed:
Merge branch 'hannibal002:beta' into block-break

CalMWolfs committed:
Merge branch 'refs/heads/beta' into fork/Thunderblade73/block-break

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/data/OtherInventoryData.kt
#	src/main/java/at/hannibal2/skyhanni/data/ScreenData.kt
#	src/main/java/at/hannibal2/skyhanni/events/InventoryCloseEvent.kt

CalMWolfs committed:
regression

CalMWolfs committed:
regression

CalMWolfs committed:
Merge branch 'beta' into block-break

hannibal002 committed:
Merge branch 'refs/heads/beta' into fork/Thunderblade73/block-break

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/features/mining/OreBlock.kt

hannibal002 committed:
fix merge conflicts

CalMWolfs committed:
Merge branch 'beta' into block-break

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/utils/InventoryUtils.kt:
```diff
@@ -131,6 +134,18 @@ object InventoryUtils {
         }
     }
 
+    fun Gui.getTitle() = when (this) {
```
please avoid using `!!`. rather use ` ?: error("bla bla")` or use the magic file somewehre to mark those fields as non null globally

Thunderblade73 committed:
no null here

Thunderblade73 committed:
Merge branch 'beta' into block-break

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/utils/InventoryUtils.kt

Thunderblade73 committed:
Merge branch 'beta' into block-break

Thunderblade73 committed:
Merge branch 'beta' into block-break

Thunderblade73 committed:
fix

Thunderblade73 committed:
Merge branch 'beta' into block-break

# Conflicts:
#	src/main/java/at/hannibal2/skyhanni/SkyHanniMod.kt
#	src/main/java/at/hannibal2/skyhanni/features/mining/OreBlock.kt

Thunderblade73 committed:
fix

Thunderblade73 committed:
Merge branch 'beta' into block-break

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/mining/BlockStrengthGuide.kt:
```diff
@@ -0,0 +1,384 @@
+                    ),
+                ),
+                listOf(
+                    Renderable.string(name.allLettersFirstUppercase()),
```
add  more info to that hover text. e.g. what does the numbers mean (the seconds, the ticks, etc)

Thunderblade73 replied:
You reference probably the old, as the screenshot above is quiet old now. Since all digits (except Block strength) are mining speed. And I think it is easily readable.
![image](https://github.com/user-attachments/assets/e262b265-7ebd-49fc-a2a7-674a66497641)

hannibal002 replied:
oh wow, yeah this looks way better!
(missing addSeparators() :P)

hannibal002 reviewed (changes requested):
i cant test this feature yet.

whatever gui i try to open, this shows up. (i press e, or i click with the sb menu in my inventory
![image](https://github.com/user-attachments/assets/47964a26-c3aa-48dd-9112-8115ec697c5a)

to reproduce: 
1. make sure the repo is updated
2. run /shblockstrength
3. open any gui

hannibal002 commented:
I assume this is bc lastAssignment is far past, therefore the check sbMenuOpened is always true?

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/mining/BlockStrengthGuide.kt:
```diff
@@ -0,0 +1,384 @@
+
+    @HandleEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        if (!sbMenuOpened) return
```
maybe the issue is this?

Thunderblade73 replied:
The issue is that the stat api is somehow broke. And the feature waits for the response of the stat api which doesn't get send, as it somehow broke

Thunderblade73 committed:
fixed Stats API

Thunderblade73 committed:
formatting hover text

Thunderblade73 committed:
added an alias (and changed it to Main)

Thunderblade73 committed:
added addSeparators

Thunderblade73 committed:
removed useless debug

Thunderblade73 committed:
fixed Header

hannibal002 committed:
Merge branch 'refs/heads/beta' into fork/Thunderblade73/block-break

hannibal002 committed:
added additional infos on hover

hannibal002 committed:
forgot

Thunderblade73 committed:
cleanup

hannibal002 committed:
small typos
